src/output/mod.rs

```diff
@@ -221,34 +228,36 @@ pub fn metrics(config: Config) {
     let file = file.unwrap();
     let mut writer = BufWriter::new(file.as_file());
 
-    let mut now = std::time::Instant::now();
-    let mut next = now + Duration::from_secs(1);
-    let end = now + config.general().duration();
+    WAIT.fetch_add(1, Ordering::Relaxed);
+
+    // get an aligned start time
+    let start = tokio::time::Instant::now() - Duration::from_nanos(Utc::now().nanosecond() as u64)
+        + Duration::from_secs(1);
+
+    // get the stop time
+    let stop = start + config.general().duration();
+
+    let mut interval = tokio::time::interval_at(start, Duration::from_secs(1));
 
     let snapshotter = SnapshotterBuilder::new()
         .metadata("source".to_string(), env!("CARGO_BIN_NAME").to_string())
         .metadata("version".to_string(), env!("CARGO_PKG_VERSION").to_string())
         .build();
 
-    while end > now {
-        std::thread::sleep(Duration::from_millis(1));
-
-        now = std::time::Instant::now();
+    while Instant::now() + config.general().interval() <= stop {
+        interval.tick().await;
 
-        if next <= now {
-            let snapshot = snapshotter.snapshot();
+        let snapshot = snapshotter.snapshot();
 
-            let buf = match config.general().metrics_format() {
-                MetricsFormat::Json => Snapshot::to_json(&snapshot).expect("failed to serialize"),
-                MetricsFormat::MsgPack | MetricsFormat::Parquet => {
-                    Snapshot::to_msgpack(&snapshot).expect("failed to serialize")
-                }
-            };
-            let _ = writer.write_all(&buf);
-
-            next += Duration::from_secs(1);
-        }
+        let buf = match config.general().metrics_format() {
+            MetricsFormat::Json => Snapshot::to_json(&snapshot).expect("failed to serialize"),
+            MetricsFormat::MsgPack | MetricsFormat::Parquet => {
+                Snapshot::to_msgpack(&snapshot).expect("failed to serialize")
+            }
+        };
+        let _ = writer.write_all(&buf);
```

Comment:
Isn't this a blocking call? I think we'll have to use `from_std` for the file on 228 and then use an `await` here.

Reply:
Good catch. I've switched to do async write and flush here.